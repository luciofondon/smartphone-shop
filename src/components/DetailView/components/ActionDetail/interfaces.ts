import { SelectChangeEvent } from '@mui/material';
import { IProductAdd } from '../../../Root/interfaces';
import { IProductDetail } from '../../interfaces';

export interface IActionDetail {
  product: IProductDetail;
  handleClickAddProduct: (product: IProductAdd) => void;
}

export interface IUseActionDetail {
  product: IProductDetail;
  handleClickAddProduct: (product: IProductAdd) => void;
}

export interface IUseActionDetailResponse {
  states: {
    storage: number;
    color: number;
  };
  actions: {
    handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
    handleChange: (event: SelectChangeEvent) => void;
  };
}
