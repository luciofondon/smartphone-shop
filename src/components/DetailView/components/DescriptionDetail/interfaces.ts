import { IProductDetail } from '../../interfaces';

export interface IDescriptionDetail {
  product: IProductDetail;
}
