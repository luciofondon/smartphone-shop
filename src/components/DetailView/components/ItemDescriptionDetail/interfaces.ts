export interface IItemDescriptrionDetail {
  label: string;
  value: string;
}
