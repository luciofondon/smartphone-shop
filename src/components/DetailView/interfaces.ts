import { IProductAdd } from '../Root/interfaces';

interface IColor {
  code: number;
  name: string;
}

interface IStorage {
  code: number;
  name: string;
}

export interface IProductDetail {
  announced: string;
  audioJack: string;
  battery: string;
  bluetooth: string[];
  brand: string;
  chipset: string;
  colors: string[];
  cpu: string;
  dimentions: string;
  displayResolution: string;
  displaySize: string;
  displayType: string;
  edge: string;
  externalMemory: string;
  gprs: string;
  gps: string;
  gpu: string;
  id: string;
  imgUrl: string;
  internalMemory: string[];
  model: string;
  networkSpeed: string;
  networkTechnology: string;
  nfc: string;
  options: { colors: IColor[]; storages: IStorage[] };
  os: string;
  price: string;
  primaryCamera: string[];
  radio: string;
  ram: string;
  secondaryCmera: string[];
  sensors: string[];
  sim: string;
  speaker: string;
  status: string;
  usb: string;
  weight: string;
  wlan: string[];
}

export interface IUseDetailViewResponse {
  states: {
    product: IProductDetail | null;
  };
  actions: {
    handleClickAddProduct: (product: IProductAdd) => void;
  };
}
