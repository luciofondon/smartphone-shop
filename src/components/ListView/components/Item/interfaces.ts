export interface IProduct {
  id: string;
  model: string;
  brand: string;
  imgUrl: string;
  price: number;
}

export interface IItem {
  product: IProduct;
}
