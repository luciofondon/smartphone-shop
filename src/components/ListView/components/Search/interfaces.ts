export interface ISearch {
  handleChangeFilter: (text: string) => void;
}
