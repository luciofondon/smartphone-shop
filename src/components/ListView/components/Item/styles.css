.item {
    padding: 60px;
    height: 200px;
    box-sizing: border-box; 
}

.image {
    height: 200px;
    width: 100px;
}




.price {
    font-size: 35px;
}

.img{
    width: 40%;
    height: 150px;
    float:left;

}

.detail{
    width: 60%;
    float: right;
}