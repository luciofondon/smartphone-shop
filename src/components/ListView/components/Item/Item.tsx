import React from 'react';
import { Link } from 'react-router-dom';

import { IItem } from './interfaces';
import './styles.css';

const Item = ({ product }: IItem) => {
  return (
    <div className="item">
      <Link to={`/detail/${product.id}`}>
        <div className="img">
          {' '}
          <img height={110} width={80} src={product.imgUrl} />
        </div>
        <div className="detail">
          <div className="model">{product.model}</div>
          <div className="brand">{product.brand}</div>
          {product.price && <div className="price"> {product.price}â‚¬</div>}
        </div>
      </Link>
    </div>
  );
};

export default Item;
