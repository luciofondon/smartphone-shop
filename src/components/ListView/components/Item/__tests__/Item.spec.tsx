import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import Item from '../Item';
import { IProduct } from '../interfaces';

test('render Item', () => {
  const productMock: IProduct = {
    brand: 'Acer',
    id: 'cGjFJlmqNPIwU59AOcY8H',
    imgUrl: 'https://front-test-api.herokuapp.com/images/cGjFJlmqNPIwU59AOcY8H.jpg',
    model: 'Liquid Z6 Plus',
    price: 220,
  };
  const { container } = render(
    <Router>
      <Item product={productMock} />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
