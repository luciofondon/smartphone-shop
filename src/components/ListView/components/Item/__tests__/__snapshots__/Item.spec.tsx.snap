// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Item 1`] = `
<div>
  <div
    class="item"
  >
    <a
      href="/detail/cGjFJlmqNPIwU59AOcY8H"
    >
      <div
        class="img"
      >
         
        <img
          height="110"
          src="https://front-test-api.herokuapp.com/images/cGjFJlmqNPIwU59AOcY8H.jpg"
          width="80"
        />
      </div>
      <div
        class="detail"
      >
        <div
          class="model"
        >
          Liquid Z6 Plus
        </div>
        <div
          class="brand"
        >
          Acer
        </div>
        <div
          class="price"
        >
           
          220
          â‚¬
        </div>
      </div>
    </a>
  </div>
</div>
`;
